### PR TITLE
improved gate decomposition

### PR DESCRIPTION
**Pull request by Simone-Bordoni**
-Optimization of FSwap gate decomposition (2 CZ instead of 4)
-RXX, RYY, RZZ do not use unitary gate decomposition (proper optimized decomposition)

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [X] Tests are passing.
- [X] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibolab/tests/test_transpilers_native.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def assert_matrices_allclose(gate):
     native_unitary = native_matrix / np.power(
         np.linalg.det(native_matrix), 1 / float(native_matrix.shape[0]), dtype=complex
     )
-    np.testing.assert_allclose(native_unitary, target_unitary, atol=1e-12)
+    np.testing.assert_allclose(phase * native_unitary, target_unitary, atol=1e-12)
```
Comment (stavros11) [inline src/qibolab/tests/test_transpilers_native.py:25]: Shouldn't this phase be removed by the det normalization?

Comment (Simone-Bordoni) [reply src/qibolab/tests/test_transpilers_native.py:25]: Unfortunately if the determinant of the matrix is +1 and the difference global phase is pi there may be problems, this happened for the FSWAP gate

Comment (stavros11) [reply src/qibolab/tests/test_transpilers_native.py:25]: Thanks.

**Comment by stavros11** [comment]
This should be good to merge.

**Pull request merged**